spec/rester/client_spec.rb

```diff
@@ -252,47 +257,105 @@ def let_retry_period_pass
+          expect(client.send(:_producer_name)).to eq "DummyService"
+        end
+      end # with producer name set
+
+      context 'with successful request' do
+        let(:logger) { double('logger') }
+        after { success_request }
+
+        it 'should log the correct messages' do
+          expect(logger).to receive(:info).with(
+            "Correlation-ID=#{Rester.correlation_id}: [Dummy] -> DummyService - GET /v1/ping"
+          ).once
+          expect(logger).to receive(:info).with(
+            "Correlation-ID=#{Rester.correlation_id}: [Dummy] <- DummyService - GET /v1/ping 200"
+          ).once
+        end
+      end # with successful request
+
+      context 'with client request_handler middleware simulation' do
+        let(:logger) { double('logger') }
+        before {
+          Rester.begin_request
+          Rester.correlation_id = SecureRandom.uuid
+        }
+        after {
+          success_request
+          Rester.end_request
+        }
+
+        it 'should log the correct messages' do
+          expect(logger).to receive(:info).with(
+            "Correlation-ID=#{Rester.correlation_id}: [Dummy] -> DummyService - GET /v1/ping"
+          ).once
+          expect(logger).to receive(:info).with(
+            "Correlation-ID=#{Rester.correlation_id}: [Dummy] <- DummyService - GET /v1/ping 200"
+          ).once
+        end
+      end # with client request_handler middleware simulation
     end # #request
 
     describe '#tests', :tests do
       let(:tests) { client.tests(*args) }
       subject { tests }
-
       context 'without connection' do
+        let(:adapter) { Client::Adapters::HttpAdapter.new }
+
         context 'with string argument' do
           let(:args) { ['token'] }
 
           describe '#get' do
             subject { tests.get }
-            it { expect { subject }.to raise_error RuntimeError, 'not connected' }
+            it { expect { subject }.to raise_error Rester::Errors::ConnectionError }
           end
 
           describe '#update' do
             subject { tests.update }
-            it { expect { subject }.to raise_error RuntimeError, 'not connected' }
+            it { expect { subject }.to raise_error Rester::Errors::ConnectionError }
           end
 
           describe '#delete' do
             subject { tests.delete }
-            it { expect { subject }.to raise_error RuntimeError, 'not connected' }
+            it { expect { subject }.to raise_error Rester::Errors::ConnectionError }
           end
         end # with string argument
 
         context 'with hash argument' do
           let(:args) { [req_hash] }
-          it { expect { subject }.to raise_error RuntimeError, 'not connected' }
+          it { expect { subject }.to raise_error Rester::Errors::ConnectionError }
         end # with hash argument
       end # without connection
 
       context 'with connection' do
-        before { client.connect(test_url) }
+        before { adapter.connect(test_url) }
```

Comment:
These tests should be moved to the adapter tests or deleted if they're duplicated.  These tests were originally testing `Client#connect` but since that method was deleted, the tests should be deleted.